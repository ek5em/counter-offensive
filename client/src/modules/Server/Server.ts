import { SHA256 } from "crypto-js";
import { io, Socket } from "socket.io-client";
import Mediator from "../Mediator/Mediator";
import Store from "../Store/Store";
import {
    IGamerInfo,
    IError,
    EGamerRole,
    IScene,
    IToken,
    IAnswer,
    ILobby,
    ETank,
    IMessage,
} from "./interfaces";
import { ESOCKET } from "../../config";

export default class Server {
    mediator: Mediator;
    HOST: string;
    STORE: Store;
    socket: Socket;

    constructor(HOST: string, mediator: Mediator) {
        this.HOST = HOST;
        this.mediator = mediator;
        this.STORE = new Store();

        this.socket = io(HOST);

        const {
            NEW_MESSAGE,
            LOGIN,
            SEND_MESSAGE_STATUS,
            LOGOUT,
            UPDATE_SCENE,
        } = mediator.getTriggerTypes();

        const { SERVER_ERROR, GO_TO_TANK, UPDATE_USER, LOBBY_UPDATE } =
            this.mediator.getEventTypes();

        this.socket.on(ESOCKET.ERROR, (answer: IError) => {
            mediator.call(SERVER_ERROR, answer.error);
        });

        this.socket.on(ESOCKET.GET_MESSAGE, (answer: IAnswer<IMessage[]>) => {
            mediator.get(NEW_MESSAGE, answer.data);
        });

        this.socket.on(ESOCKET.SEND_MESSAGE, (answer: IAnswer<true>) => {
            mediator.get(SEND_MESSAGE_STATUS, answer.data);
        });

        this.socket.on(ESOCKET.REGISTRATION, (answer: IAnswer<IToken>) => {
            mediator.get(LOGIN, answer.data);
        });

        this.socket.on(ESOCKET.LOGIN, (answer: IAnswer<IToken>) => {
            mediator.get(LOGIN, answer.data);
        });

        this.socket.on(ESOCKET.GET_USER_INFO, (answer: IAnswer<IGamerInfo>) => {
            this.STORE.setUser(answer.data);
            mediator.call(UPDATE_USER, answer.data);
        });

        this.socket.on(ESOCKET.LOGOUT, () => {
            mediator.get(LOGOUT);
        });

        this.socket.on(ESOCKET.TOKEN_VERIFICATION, () => {
            mediator.get(LOGIN, { token: this.STORE.getToken() });
        });

        this.socket.on(
            ESOCKET.SET_GAMER_ROLE,
            (answer: IAnswer<{ tankId: number; tankType: ETank }>) => {
                if (answer.data.tankId) {
                    mediator.call(GO_TO_TANK, answer.data);
                }
            }
        );

        this.socket.on(ESOCKET.GET_LOBBY, (answer: IAnswer<ILobby>) => {
            this.STORE.setLobby(answer.data);
            mediator.call(LOBBY_UPDATE, answer.data);
        });

        this.socket.on(ESOCKET.GET_SCENE, (answer: IAnswer<IScene>) => {
            mediator.get(UPDATE_SCENE, answer.data);
        });
    }

    // АВТОРИЗАЦИЯ

    registration(login: string, nickname: string, password: string) {
        const hash = SHA256(login + password).toString();
        this.socket.emit(ESOCKET.REGISTRATION, { login, nickname, hash });
    }

    login(login: string, password: string) {
        const rnd = Math.random();
        const hash = SHA256(
            SHA256(login + password).toString() + rnd
        ).toString();
        this.socket.emit(ESOCKET.LOGIN, { login, hash, rnd });
    }

    tokenVerification() {
        this.socket.emit(ESOCKET.TOKEN_VERIFICATION, {
            token: this.STORE.getToken(),
        });
    }

    getUser() {
        this.socket.emit(ESOCKET.GET_USER_INFO, {
            token: this.STORE.getToken(),
        });
    }

    logout() {
        this.socket.emit(ESOCKET.LOGOUT, {
            token: this.STORE.getToken(),
        });
    }

    // ЛОББИ

    sendMessage(message: string) {
        this.socket.emit(ESOCKET.SEND_MESSAGE, {
            token: this.STORE.getToken(),
            message,
        });
    }

    getMessages() {
        this.socket.emit(ESOCKET.GET_MESSAGE, {
            token: this.STORE.getToken(),
        });
    }

    setGamerRole(role: EGamerRole, tankId: number | null = null) {
        this.socket.emit(ESOCKET.SET_GAMER_ROLE, {
            token: this.STORE.getToken(),
            role,
            tankId,
        });
    }

    getLobby() {
        this.socket.emit(ESOCKET.GET_LOBBY, {
            token: this.STORE.getToken(),
        });
    }

    // ИГРА

    getScene() {
        this.socket.emit(ESOCKET.GET_SCENE, {
            token: this.STORE.getToken(),
        });
    }

    suicide() {
        this.socket.emit(ESOCKET.SUICIDE, {
            token: this.STORE.getToken(),
        });
    }

    unitMotion(x: number | null, y: number | null, angle: number) {
        this.socket.emit(ESOCKET.MOTION, {
            token: this.STORE.getToken(),
            x,
            y,
            angle,
        });
    }

    makeShot(x: number, y: number, angle: number) {
        this.socket.emit(ESOCKET.FIRE, {
            token: this.STORE.getToken(),
            x,
            y,
            angle,
        });
    }
}
