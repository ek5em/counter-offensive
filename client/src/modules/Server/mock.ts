import {
    EGamerRole,
    ERank,
    IGamerInfo,
    ILobby,
    IMessage,
    IUserUnit,
} from "./interfaces";

interface IMock {
    users: (IGamerInfo & { password: string })[];
    messages: {
        text: string;
        rank_name: ERank;
        userId: number;
        nickname: string;
    }[];
    lobby: ILobby;
    unit: IUserUnit;
}

export const MOCK: IMock = {
    users: [
        {
            gamer_exp: 9999,
            id: 0,
            is_alive: null,
            level: 10,
            login: "123",
            next_rank: 65421,
            nickname: "Гришка",
            password: "123",
            rank_name: ERank.General,
            token: "123",
        },
        {
            gamer_exp: 9999,
            id: 1,
            is_alive: null,
            level: 10,
            login: "321",
            next_rank: 65421,
            nickname: "Петька заводной",
            password: "321",
            rank_name: ERank.General,
            token: "321",
        },
    ],
    messages: [
        {
            nickname: "Гришка",
            rank_name: ERank.General,
            text: "Привет, парни!",
            userId: 0,
        },
        {
            nickname: "Петька заводной",
            rank_name: ERank.General,
            text: "Привет, Гришка, ну чо как вчера жёско зарубимся в контр-нахрюк???!",
            userId: 1,
        },
    ],
    lobby: {
        bannerman: true,
        general: true,
        tanks: {
            heavyTank: [
                { id: 1, Commander: true, Gunner: false, Mechanic: false },
                { id: 3, Commander: false, Gunner: true, Mechanic: false },
            ],
            middleTank: [{ id: 2, Gunner: true, Mechanic: false }],
        },
    },
    unit: {
        angle: 0,
        personId: EGamerRole.infantry,
        speed: 1,
        x: 5,
        y: 5,
    },
};
