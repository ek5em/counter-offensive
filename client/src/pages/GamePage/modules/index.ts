import Collision from "./Game/Collision/Collision";
import MathGame from "./Game/Math/MathGame";

import Game from "./Game/Game";
import Canvas from "./Graph/Canvas/Canvas";
import TraceMask from "./Graph/TraceMask";

export { Canvas, Collision, Game, TraceMask, MathGame };

export * from "./Game/Units";
