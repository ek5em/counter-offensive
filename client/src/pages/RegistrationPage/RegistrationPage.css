.reg_header {
  display: flex;
  justify-content: space-between;
  height: 20%;
  gap:20px;
}

.reg_header button {
  font-size: 60px;
  padding: 0 10px; 
}

.reg_wrapper {
  height: 100%;
  width: 100%;
}

.reg_content {
  display: flex;
  flex-direction: column;
  height: 80%;
}

.reg_submit_button {
  font-size: 60px;
  margin: 10px 0;
}

.reg_form {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
}

.reg_footer {
  display: flex;
  justify-content: flex-end;
}

@media  (max-width: 1130px) {

  .reg_header button,
  .reg_submit_button  {
    font-size: 45px;
  }
  
}

@media  (max-width: 750px) {
  .reg_header button {
    font-size: 40px;
    padding: 0 10px;
  }
  
}

@media (max-width: 550px) {
  .reg_header button {
    font-size: 30px;
    padding: 0 10px;
  }

  .reg_submit_button{
    font-size: 25px;
    padding: 0 10px;
  }
  
}

@media (max-width: 353px) {
  .reg_header button {
    font-size: 25px;
  }
}
