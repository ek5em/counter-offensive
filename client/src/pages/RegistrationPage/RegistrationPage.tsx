import { FC } from "react";
import { Button, Registration } from "../../components";
import "./RegistrationPage.css";
import { Link } from "react-router-dom";

const RegistrationPage: FC = () => {
  return (
    <div className="reg_wrapper">
      <div className="reg_header">
        <div>
          <Button appearance="primary-disable" id="test_reg_pageName_button">
            Повестка
          </Button>
        </div>
        <div>
          <Link to='/authorization'>
          <Button appearance="primary" id="test_reg_goToMain_button">
            Уже служил
          </Button>
          </Link>
        </div>
      </div>
      <div className="reg_content">
        <Registration />
      </div>
    </div>
  );
};

export default RegistrationPage;
