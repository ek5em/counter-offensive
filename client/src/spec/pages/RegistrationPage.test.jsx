import { fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import {
    ContextProvider,
    MediatorContext,
    ServerContext,
} from "../../components";
import { MOCK } from "../../modules/Server/mock";
import { Mediator, Server } from "../../modules";
import { MEDIATOR, HOST } from "../../config";
import { RegistrationPage } from "../../pages/AuthPages/RegistrationPage/RegistrationPage";

describe("Страничка регистрации", () => {
    const getLoginInput = () => screen.getByPlaceholderText(/логин/i);
    const getNicknameInput = () => screen.getByPlaceholderText(/никнейм/i);
    const getPasswordInput = () => screen.getByPlaceholderText(/пароль/i);
    const getSubmitBtn = () =>
        screen.getByRole("button", { name: /подписать повестку/i });
    const getNavBtn = () => screen.getByRole("button", { name: /уже служил/i });

    const createFields = ({
        login = "11111111",
        password = "11111111",
        nickname = "superJaba123",
    } = {}) => {
        render(
            <ContextProvider useMock={true}>
                <RegistrationPage />
            </ContextProvider>
        );

        fireEvent.change(getLoginInput(), { target: { value: login } });
        fireEvent.change(getPasswordInput(), { target: { value: password } });
        fireEvent.change(getNicknameInput(), { target: { value: nickname } });
        fireEvent.click(getSubmitBtn());
    };

    test("Наличиие всех элементов", () => {
        const { asFragment } = render(
            <ContextProvider>
                <RegistrationPage />
            </ContextProvider>
        );
        expect(getLoginInput()).toBeInTheDocument();
        expect(getNicknameInput()).toBeInTheDocument();
        expect(getPasswordInput()).toBeInTheDocument();
        expect(getSubmitBtn()).toBeInTheDocument();
        expect(getNavBtn()).toBeInTheDocument();
        
        expect(asFragment()).toMatchSnapshot();
    });

    test("Логин меньше 6 символов", () => {
        createFields({ login: "a" });
        expect(
            screen.getByText(/Логин должен содержать от 6 до 15 символов/i)
        ).toBeInTheDocument();
    });

    test("Логин больше 15 символов", () => {
        createFields({ login: "1111111111111111" });
        expect(
            screen.getByText(/Логин должен содержать от 6 до 15 символов/i)
        ).toBeInTheDocument();
    });

    test("Неверные символы логин", () => {
        createFields({ login: "abcdefg::" });
        expect(
            screen.getByText(
                /Логин может содержать символы кириллицы, латиницы и цифры/i
            )
        ).toBeInTheDocument();
    });

    test("Неверные символы других языков логин", () => {
        createFields({ login: "ざざざざざざざざ" });
        expect(
            screen.getByText(
                /Логин может содержать символы кириллицы, латиницы и цифры/i
            )
        ).toBeInTheDocument();
    });

    test("Логин занят", () => {
        createFields({ ...MOCK.users[0] });
        expect(screen.getByText(/Логин занят/i)).toBeInTheDocument();
    });

    test("Никнейм короче 3 символов", () => {
        createFields({ nickname: "a" });
        expect(
            screen.getByText(/Никнейм должен содержать от 3 до 16 символов/i)
        ).toBeInTheDocument();
    });

    test("Никнейм длиннее 16 символов", () => {
        createFields({ nickname: "111111111111111111" });
        expect(
            screen.getByText(/Никнейм должен содержать от 3 до 16 символов/i)
        ).toBeInTheDocument();
    });

    test("Никнейм содержит неверные символы", () => {
        createFields({ nickname: "asd:;" });
        expect(
            screen.getByText(
                /Никнейм может содержать символы любого языка и цифры/i
            )
        ).toBeInTheDocument();
    });

    test("Пароль меньше 8 символов", () => {
        createFields({ password: "aa" });
        expect(
            screen.getByText(/В пароле должно быть от 8 до 20 символов/i)
        ).toBeInTheDocument();
    });

    test("Пароль больше 20 символов", () => {
        createFields({ password: "aaaaaaaaaaaaaaaaaaaaaa" });
        expect(
            screen.getByText(/В пароле должно быть от 8 до 20 символов/i)
        ).toBeInTheDocument();
    });

    test("Верные данные", () => {
        createFields();
        expect(
            screen.queryByText(
                /Логин может содержать символы кириллицы, латиницы и цифры/i
            )
        ).not.toBeInTheDocument();
        expect(
            screen.queryByText(/Логин должен содержать от 6 до 15 символов/i)
        ).not.toBeInTheDocument();

        expect(
            screen.queryByText(/Никнейм должен содержать от 3 до 16 символов/i)
        ).not.toBeInTheDocument();
        expect(
            screen.queryByText(
                /Никнейм может содержать символы любого языка и цифры/i
            )
        ).not.toBeInTheDocument();

        expect(
            screen.queryByText(/В пароле должно быть от 8 до 20 символов/i)
        ).not.toBeInTheDocument();
    });

    /* test("Регистрация", () => {
        const mediator = new Mediator(MEDIATOR);
        const server = new Server(HOST, mediator, true);
        const registrationSpy = jest
            .spyOn(server, "registration")
            .mockImplementation(jest.fn());

        render(
            <BrowserRouter>
                <MediatorContext.Provider value={mediator}>
                    <ServerContext.Provider value={server}>
                        <RegistrationPage />
                    </ServerContext.Provider>
                </MediatorContext.Provider>
            </BrowserRouter>
        );
        // Ввод данных
        fireEvent.change(getLoginInput(), { value: login });
        fireEvent.change(getPasswordInput(), { value: password });
        fireEvent.change(getNicknameInput(), { value: nickname });

        fireEvent.click(getSubmitBtn());
        // expect(registrationSpy).toHaveBeenCalledWith(login, nickname, password);
        // expect(registrationSpy).toHaveBeenCalledTimes(1);
    }); */
});
