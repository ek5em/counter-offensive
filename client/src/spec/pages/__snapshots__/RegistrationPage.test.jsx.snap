// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Страничка регистрации Верные данные 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert disabled"
            id=""
          >
            <span />
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Логин больше 15 символов 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_login_length"
          >
            <span>
              Логин должен содержать от 6 до 15 символов
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Логин занят 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert error"
            id="test_error_reg_loginOccupied"
          >
            <span>
              Логин занят
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Логин меньше 6 символов 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_login_length"
          >
            <span>
              Логин должен содержать от 6 до 15 символов
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Неверные символы других языков логин 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_acceptableSymbolsLogin"
          >
            <span>
              Логин может содержать символы кириллицы, латиницы и цифры
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Неверные символы логин 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_acceptableSymbolsLogin"
          >
            <span>
              Логин может содержать символы кириллицы, латиницы и цифры
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Никнейм длиннее 16 символов 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_nickname_length"
          >
            <span>
              Никнейм должен содержать от 3 до 16 символов
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Никнейм короче 3 символов 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_nickname_length"
          >
            <span>
              Никнейм должен содержать от 3 до 16 символов
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Никнейм содержит неверные символы 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_acceptableSymbolsNickname"
          >
            <span>
              Никнейм может содержать символы любого языка и цифры
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Пароль больше 20 символов 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_password_length"
          >
            <span>
              В пароле должно быть от 8 до 20 символов
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Страничка регистрации Пароль меньше 8 символов 1`] = `
<DocumentFragment>
  <div
    class="authPage"
  >
    <div
      class="logo"
    >
      <a
        href="/"
      >
        <img
          alt="counter-nastup"
          src="Logo.png"
        />
      </a>
    </div>
    <div
      class="wrapper"
    >
      <div
        class="content"
      >
        <div
          class="title"
          id="test_auth_reg_title"
        >
          <a
            href="/authorization"
            tabindex="-1"
          >
            <button
              class="button auth_switch_page primary"
              id="test_login_go_log_button"
            >
              Уже служил
            </button>
          </a>
        </div>
        <form
          class="form"
        >
          <div
            class="values"
          >
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_log_input"
                placeholder="Логин"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input"
                id="test_reg_nick_input"
                placeholder="Никнейм"
                type="text"
              />
            </div>
            <div
              class="wrapper"
            >
              <input
                autocomplete="off"
                class="input password"
                id="test_reg_pass_input"
                placeholder="Пароль"
                type="password"
              />
              <div
                class="eye"
              >
                <svg
                  data-testid="close-eye-icon"
                >
                  closeEye.svg
                </svg>
              </div>
            </div>
          </div>
          <div
            class="alert warning"
            id="test_warning_reg_password_length"
          >
            <span>
              В пароле должно быть от 8 до 20 символов
            </span>
          </div>
          <button
            class="button submit primary"
            id="test_reg_submit_button"
          >
            Подписать повестку
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
